### ISSUE TITLE
[Question]: Should `teal` codebase use `magrittr`

### ISSUE THREAD

**Issue by m7pr** [insightsengineering/teal#1268]
Currently `magrittr` pipe is reexported https://github.com/insightsengineering/teal/blob/f0e3969141abb97587e696dbb8a57ad081949173/R/teal.R#L14

The motivation for that is probably for the users to be able to use pipe, after teal is attached. Should we use `%>%` in the codebase, since it's already in Imports, or should we reconsider cleaning this up?

**Comment by chlebowa** [comment]
Related issue: https://github.com/insightsengineering/teal.modules.general/issues/786

**Comment by m7pr** [comment]
I agree with the removal of `%>%` from the codebase, but I'm not sure we really rely on `4.1`

**Comment by chlebowa** [comment]
`teal` `Depends` on R 4.0.

**Comment by averissimo** [comment]
`{teal}`'s current dependencies depend on `4.4` due to `{Matrix}` package, but I think there's no real technical requirement to force us to bump to `4.1`

Users using a CRAN snapshot from before the 4.4 release date can probably use 4.0.

**Comment by donyunardi** [comment]
The teal codebase and its framework packages no longer use `|>` or `%>%`. We have also removed the `%>%` dependency mentioned in the original post.

So, the consensus is to keep the R version dependency for teal at 4.0.

However, [tmg](https://github.com/insightsengineering/teal.modules.general/blob/643644eb0d839c755cc796f6074b6f8c4b9b59ea/R/teal.modules.general.R#L11) and [tmc](https://github.com/insightsengineering/teal.modules.clinical/blob/af36c1740e9016cab0fbf3fd9d48959472a7d90c/R/teal.modules.clinical.R#L12) are still uses `%>%`.

We could consider removing this dependency from the teal module packages that we maintain.
Additionally, please be mindful of the behavior of `|>` in the teal_data/qenv context on code reproducibility.
![image](https://github.com/user-attachments/assets/d571271b-f776-450d-bdcc-407bfba0562d)

As you see, the code is updated to not use the `|>` operator.

And here is example if we use `%>%` operator where code stored with the operator.
![image](https://github.com/user-attachments/assets/f43503a2-c997-46a4-84c2-658b92e50d02)

**Comment by gogonzo** [comment]
Closing this as @averissimo made it clear in his comments
